space/components/accounts/sign-in-forms/email-form.tsx

```diff
@@ -0,0 +1,148 @@
+import React, { useState } from "react";
+import { Controller, useForm } from "react-hook-form";
+import { XCircle } from "lucide-react";
+// services
+import authService from "services/authentication.service";
+// hooks
+import useToast from "hooks/use-toast";
+// ui
+import { Button, Input } from "@plane/ui";
+// helpers
+import { checkEmailValidity } from "helpers/string.helper";
+// types
+import { IEmailCheckData, TEmailCheckTypes } from "types/auth";
+// constants
+import { ESignInSteps } from "components/accounts";
+
+type Props = {
+  handleStepChange: (step: ESignInSteps) => void;
+  updateEmail: (email: string) => void;
+};
+
+type TEmailCodeFormValues = {
+  email: string;
+};
+
+export const EmailForm: React.FC<Props> = (props) => {
+  const { handleStepChange, updateEmail } = props;
+  // states
+  const [isCheckingEmail, setIsCheckingEmail] = useState<TEmailCheckTypes | null>(null);
+
+  const { setToastAlert } = useToast();
+
+  const {
+    control,
+    formState: { errors, isValid },
+    handleSubmit,
+    watch,
+  } = useForm<TEmailCodeFormValues>({
+    defaultValues: {
+      email: "",
+    },
+    mode: "onChange",
+    reValidateMode: "onChange",
+  });
+
+  const handleEmailCheck = async (type: TEmailCheckTypes) => {
+    setIsCheckingEmail(type);
+
+    const email = watch("email");
+
+    const payload: IEmailCheckData = {
+      email,
+      type,
+    };
+
+    // update the global email state
+    updateEmail(email);
+
+    await authService
+      .emailCheck(payload)
+      .then((res) => {
+        // if type is magic_code, send the user to magic sign in
+        if (type === "magic_code") handleStepChange(ESignInSteps.UNIQUE_CODE);
+        // if type is password, check if the user has a password set
+        if (type === "password") {
+          // if password is autoset, send them to set new password link
+          if (res.is_password_autoset) handleStepChange(ESignInSteps.SET_PASSWORD_LINK);
+          // if password is not autoset, send them to password form
+          else handleStepChange(ESignInSteps.PASSWORD);
+        }
+      })
+      .catch((err) =>
+        setToastAlert({
+          type: "error",
+          title: "Error!",
+          message: err?.error ?? "Something went wrong. Please try again.",
+        })
+      )
+      .finally(() => setIsCheckingEmail(null));
+  };
+
+  return (
+    <>
+      <h1 className="text-center text-2xl font-semibold text-onboarding-text-100">Get on your flight deck!</h1>
```

Comment:
Update the content here.

Reply:
I've implemented the content provided. If there are any further updates or changes you'd like to include, please let me know. I'll make sure to review and implement any additional changes as needed.